File: logic-test.js
function sortString(str) {
    return str.split('').sort().join('');
}

function groupAnagrams(words) {
    var anagramMap = {};

    for (var i = 0; i < words.length; i++) {
        var word=words[i];
        var sortedWord = sortString(word);
        if (!anagramMap[sortedWord]) {
            anagramMap[sortedWord] = [];
        }
        anagramMap[sortedWord].push(word);
    }

    var result = [];
    for(var key in anagramMap) {
        result.push(anagramMap[key]);
    }
    return result;
}

var words = ['cook', 'save', 'taste', 'aves', 'vase', 'state', 'map'];
var anagramGroups = groupAnagrams(words);
console.log(anagramGroups);

File: LogicTest.java
import java.util.*;

public class LogicTest {
    private static List<List<String>> groupAnagrams(String[] words) {
        Map<String, List<String>> anagrams = new HashMap<>();

        for (String word : words) {
            char[] chars = word.toCharArray();
            Arrays.sort(chars);
            String sortedWord = new String(chars);

            if (!anagrams.containsKey(sortedWord)) {
                anagrams.put(sortedWord, new ArrayList<>());
            }
            anagrams.get(sortedWord).add(word);
        }

        return new ArrayList<>(anagrams.values());
    }
    private static String sortString(String str) {
        char[] chars = str.toCharArray();
        Arrays.sort(chars);
        return new String(chars);
    }

    public static void main(String[] args) {
        String[] words = {"cook", "save", "taste", "aves", "vase", "state", "map"};
        List<List<String>> anagrams = groupAnagrams(words);
        System.out.println(anagrams);
    }
}


